### ISSUE TITLE
CUDA problem

### ISSUE THREAD

**Issue by mmagithub** [choderalab/yank#1235]
Hi,
I am wondering if there is a specific recipe to follow to install and run YANK (Anaconda) with CUDA enabled. It seems that the different CUDA / openmm / pymbar versions create several issues and inconsistencies.
I have tried to install the most recent version as outlined on the web page I keep getting this error:
############
return _openmm.Platform_getPlatformByName(name)
Exception: There is no registered Platform called "CUDA"
##############
I tried to load/unload different modules, upgrade or downgrade YANK or openmm, trying different CUDA versions, setting the plugin / nvcc, but no success?
Any recommendations?

Thanks,
Marawan

**Comment by mmagithub** [comment]
I have used the default installation protocol (through Anaconda3) as documented on the website, which I beleieve should install all required accessory software (Openmm, pymbar, armed, etc). On the cluster I have different versions of CUDA (from CUDA8 to CUDA10), I tried to load or specify the path directly (nvcc and lib64) but this did not solve the issue.
The YANK platforms command outputs only CPU as an available platform.

**Comment by mmagithub** [comment]
This is how I run it:

source  /home/mmahmed/anaconda3/bin/activate
export PATH="/home/mmahmed/anaconda3/bin:$PATH"

export PATH=/cvmfs/soft.computecanada.ca/easybuild/software/2017/avx2/Compiler/intel2016.4/cuda/10.0.130/bin:$PATH
export OPENMM_CUDA_COMPILER=/cvmfs/soft.computecanada.ca/easybuild/software/2017/avx2/Compiler/intel2016.4/cuda/10.0.130/bin/nvcc
export LD_LIBRARY_PATH=/cvmfs/soft.computecanada.ca/easybuild/software/2017/avx2/Compiler/intel2016.4/cuda/10.0.130/lib64
export CUDA_HOME=/cvmfs/soft.computecanada.ca/easybuild/software/2017/avx2/Compiler/intel2016.4/cuda/10.0.130/
export CUDA_INCLUDE_DIR=/cvmfs/soft.computecanada.ca/easybuild/software/2017/avx2/Compiler/intel2016.4/cuda/10.0.130/include
export CUDA_LIB_DIR=/cvmfs/soft.computecanada.ca/easybuild/software/2017/avx2/Compiler/intel2016.4/cuda/10.0.130/lib64
export LD_LIBRARY_PATH=/cvmfs/soft.computecanada.ca/easybuild/software/2017/avx2/Compiler/intel2016.4/cuda/10.0.130/lib64
export OPENMM_PLUGIN_DIR=~/anaconda3/lib/plugins

export PREFIX="explicit_ima"
build_mpirun_configfile --configfilepath $PREFIX.configfile "yank script --yaml=$PREFIX.yaml"
mpiexec -configfile $PREFIX.configfile

and this is the error i get:

Exception: There is no registered Platform called "CUDA"
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 1
- [ ]

**Comment by andrrizzi** [comment]
Hi @mmagithub . What happens if you install only OpenMM through conda? Can OpenMM see the CUDA platform?

**Comment by mmagithub** [comment]
Hi Andy, interesting, I will try that.
As a side note, after I gave up, I tried to downgrade to YANK 0.23.7 and it is currently running, it gave some scipy related errors that I had o fix manually within the multistate analysis scripts (I think there was a confusion between spatial and misc modules). If it finishes correctly, I think I will stick to this version for a while, perhaps the YANK 0.25.2 version (the current default recipe from Anaconda) still have some issue. If I found other problems down the road, I will try your suggestions and see how it goes.
Have a wonderful day
